pants.toml

```diff
@@ -79,3 +80,12 @@ args = [
   # needed in st2* components, runners
   "--load-plugins=db_models",
 ]
+
+[subprocess-environment]
+env_vars.add = [
+  # Works around bad `-arch arm64` flag embedded in Xcode 12.x Python interpreters on intel
+  # machines. See: https://github.com/giampaolo/psutil/issues/1832
+  # On macos do:  export ARCHFLAGS=-arch x86_64 
+  # see: https://github.com/pantsbuild/pants/pull/11733
+  "ARCHFLAGS",
```

Comment:
When I hit this issue, I switched over to my MacPorts-installed copies of python. Good to know there's a solution :)

This won't affect any other platforms will it?

Reply:
this option just tells pants to pass this env variable (if defined) to subprocesses. it doesn't actually define it.
in CI, pants defines it for mac os builds, see https://github.com/pantsbuild/pants/pull/11733

Reply:
and there is a thread discussion in pants slack: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1617914862144100